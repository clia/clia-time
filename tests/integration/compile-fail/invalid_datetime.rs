use time::macros::datetime;

fn main() {
    let _ = datetime!(2021 - 000 0:00);
    let _ = datetime!(2021 - 001 24:00);
    let _ = datetime!(2021 - 001 0:00 0);
    let _ = datetime!(2021 - 001 0:00 UTC x);
}
