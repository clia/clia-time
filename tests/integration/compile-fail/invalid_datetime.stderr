error: invalid component: ordinal was 0
 --> $DIR/invalid_datetime.rs:4:30
  |
4 |     let _ = datetime!(2021 - 000 0:00);
  |                              ^^^

error: invalid component: hour was 24
 --> $DIR/invalid_datetime.rs:5:34
  |
5 |     let _ = datetime!(2021 - 001 24:00);
  |                                  ^^

error: unexpected token: 0
 --> $DIR/invalid_datetime.rs:6:39
  |
6 |     let _ = datetime!(2021 - 001 0:00 0);
  |                                       ^

error: unexpected token: x
 --> $DIR/invalid_datetime.rs:7:43
  |
7 |     let _ = datetime!(2021 - 001 0:00 UTC x);
  |                                           ^
